ssl/t1_lib.c

```diff
@@ -417,6 +417,320 @@ int ssl_load_groups(SSL_CTX *ctx)
+            }
+            else {
+                if (sinf->sig_name)
+                    keytype_nid = OBJ_txt2nid(sinf->sig_name);
+                else
+                    keytype_nid = OBJ_txt2nid(sinf->sigalg_name);
+            }
+	    if (sinf->hash_name) {
+                OBJ_create(sinf->hash_oid, sinf->hash_name, NULL);
+                if (sinf->sig_name)
+                    OBJ_add_sigid(OBJ_txt2nid(sinf->sig_name),
+                                  OBJ_txt2nid(sinf->hash_name),
+                                  keytype_nid);
+            }
+	    else {
+                OBJ_add_sigid(OBJ_txt2nid(sinf->sigalg_name),
+                              NID_undef,
+                              keytype_nid);
+            }
+	    /* sanity check: Without successful registration don't use alg */
+	    if ((OBJ_txt2nid(sinf->sigalg_name) == NID_undef) ||
+                (OBJ_nid2obj(OBJ_txt2nid(sinf->sigalg_name)) == NULL)) {
+                    ERR_raise(ERR_LIB_SSL, ERR_R_PASSED_INVALID_ARGUMENT);
+		    goto err;
+	    }
+            ctx->sigalg_list_len++;
+            sinf = NULL;
+        }
+        EVP_KEYMGMT_free(keymgmt);
+    }
+    ERR_pop_to_mark();
+ err:
+    if (sinf != NULL) {
+        OPENSSL_free(sinf->name);
+        sinf->name = NULL;
+        OPENSSL_free(sinf->sigalg_name);
+        sinf->sigalg_name = NULL;
+        OPENSSL_free(sinf->sigalg_oid);
+        sinf->sigalg_oid = NULL;
+        OPENSSL_free(sinf->sig_name);
+        sinf->sig_name = NULL;
+        OPENSSL_free(sinf->sig_oid);
+        sinf->sig_oid = NULL;
+        OPENSSL_free(sinf->hash_name);
+        sinf->hash_name = NULL;
+        OPENSSL_free(sinf->hash_oid);
+        sinf->hash_oid = NULL;
+        OPENSSL_free(sinf->keytype);
+        sinf->keytype = NULL;
+        OPENSSL_free(sinf->keytype_oid);
+        sinf->keytype_oid = NULL;
+    }
+    return ret;
+}
+
+static int discover_provider_sigalgs(OSSL_PROVIDER *provider, void *vctx)
+{
+    struct provider_ctx_data_st pgd;
+
+    pgd.ctx = vctx;
+    pgd.provider = provider;
+    OSSL_PROVIDER_get_capabilities(provider, "TLS-SIGALG",
+                                   add_provider_sigalgs, &pgd);
+    /*
+     * Always OK, even if provider doesn't support the capability:
+     * Reconsider testing retval when legacy sigalgs are also loaded this way.
+     */
+    return 1;
+}
+
+int ssl_load_sigalgs(SSL_CTX *ctx)
+{
+    size_t i;
+    SSL_CERT_LOOKUP lu;
+
+    if (!OSSL_PROVIDER_do_all(ctx->libctx, discover_provider_sigalgs, ctx))
+        return 0;
+
+    /* now populate ctx->ssl_cert_info */
+    if (ctx->sigalg_list_len > 0) {
+        ctx->ssl_cert_info = OPENSSL_zalloc(sizeof(lu) * ctx->sigalg_list_len);
+        if (ctx->ssl_cert_info == NULL)
+            return 0;
+        for(i = 0; i < ctx->sigalg_list_len; i++) {
+            ctx->ssl_cert_info[i].nid = OBJ_sn2nid(ctx->sigalg_list[i].sigalg_name);
```

Comment:
Please use `OBJ_txt2nid`

Reply:
so changed in https://github.com/openssl/openssl/pull/19312/commits/1f20a63fbcb8f733651546e40328ae2ab2090537